server/api/crud/model_monitoring/deployment.py

```diff
@@ -1203,6 +1211,41 @@ def set_credentials(
             ),
         )
 
+    def _is_the_same_cred(
+        self, endpoint_store_connection: str, stream_path: str, tsdb_connection: str
+    ) -> bool:
+        credentials_dict = {
+            key: server.api.crud.Secrets().get_project_secret(
+                project=self.project,
+                provider=mlrun.common.schemas.SecretProviderName.kubernetes,
+                secret_key=key,
+                allow_secrets_from_k8s=True,
+            )
+            for key in mlrun.common.schemas.model_monitoring.ProjectSecretKeys.mandatory_secrets()
+        }
+
+        if (
+            credentials_dict[
+                mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ENDPOINT_STORE_CONNECTION
+            ]
+            != endpoint_store_connection
+        ):
+            return False
```

Comment:
another recommendation - add a log before the `False` just to mention the specific secret that has a conflict. It will be useful for debugging as well (e.g. understanding that there is an issue with the store connection secret).